pkg/cmd/reset.go

```diff
@@ -100,5 +105,43 @@ func runReset(opts *resetOpts) error {
 
 	s.Logger.Warnln("this command will require an explicit confirmation starting with the next minor release (v1.3)")
 
+	fmt.Println("The following nodes will be resetted: ")
+	for _, node := range s.Cluster.ControlPlane.Hosts {
+		fmt.Printf("\t+ reset control plane node %q (%s)\n", node.Hostname, node.PrivateAddress)
+	}
+	for _, node := range s.Cluster.StaticWorkers.Hosts {
+		fmt.Printf("\t+ reset static worker nodes %q (%s)\n", node.Hostname, node.PrivateAddress)
+	}
+
+	if !s.Cluster.MachineController.Deploy {
+		s.Logger.Info("Skipping deleting workers because machine-controller is disabled in configuration.")
+	}
+
+	if s.DynamicClient == nil {
+		return errors.New("kubernetes client not initialized")
+	}
```

Comment:
It's mostly like that this will always be `nil`. If it's `nil`, you should initialize the DynamicClient by calling the following function: https://github.com/kubermatic/kubeone/blob/af4f8c57d1b5d6fabd01c9694422d2347349274f/pkg/kubeconfig/kubeconfig.go#L52-L74

Reply:
@xmudrii  has been something like this in your mind? https://github.com/kubermatic/kubeone/pull/1251/commits/c0022c072e866a06d3acbce021d26478164c0b33